crates/ruff_python_formatter/src/builders.rs

```diff
@@ -31,20 +31,32 @@ impl ParenthesizeIfExpands<'_, '_> {
 impl<'ast> Format<PyFormatContext<'ast>> for ParenthesizeIfExpands<'_, 'ast> {
     fn fmt(&self, f: &mut Formatter<PyFormatContext<'ast>>) -> FormatResult<()> {
         {
+            let parens_id = f.group_id("optional_parentheses");
```

Comment:
We should avoid creating the id if `self.indent` is false. We should also use a better unique name